src/main/java/com/example/demo/services/BooksServiceImpl.java

```diff
@@ -0,0 +1,47 @@
+package com.example.demo.services;
+
+import com.example.demo.model.Book;
+import com.example.demo.repository.BookRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+@Service
+public class BooksServiceImpl implements BookService {
+
+    private BookRepository bookRepository;
+
+    @Autowired
+    BooksServiceImpl(BookRepository bookRepository) {
+        this.bookRepository = bookRepository;
+    }
+
+    @Override
+    public List<Book> findAll() {
+        return bookRepository.findAll();
+    }
+
+    public List<Book> findBooksContainingWord(String word) {
+        Stream<Book> books = booksContainingWord(word);
+        return books.collect(Collectors.toList());
+    }
+
+    private Stream<Book> booksContainingWord(String word) {
+        Stream<Book> bookStream = findAll().stream();
+        return bookStream.filter(book -> isBookContainingWord(book, word));
+    }
+
+    private boolean isBookContainingWord(Book book, String word) {
+        String bookInfo = bindBookFields(book);
+        return bookInfo.contains(word);
```

Comment:
Jezeli bede szukał słowa "dog", a bede mial autora "Gido" i język "german", to zakwalifikuje dane ksiazke, a nie powinnienem. Coś tu tu nie gra. It's a bug!